src/git.rs

```diff
@@ -203,14 +207,25 @@ impl Repository {
                         ]
                         .iter(),
                     );
-                url.set_query(Some(&format!("ref={}", tag)));
+                url.query_pairs_mut().append_pair("ref", tag);
                 if let Some(token) = private_token {
-                    url.set_query(Some(&format!("private_token={}", token)));
+                    url.query_pairs_mut().append_pair("private_token", token);
                 }
                 Some(url)
             },
         })
     }
+
+    fn filename(&self) -> Option<String> {
+        match self {
+            Repository::GitLab { repo_path, .. } => Some(format!(
+                "npins-gitlab-{}.tar.gz",
+                repo_path.split('/').last().unwrap().to_owned()
```

Comment:
What are our assurances that we can really unwrap here? Do we verify this while unserializing? Do we verify it (only?) during interactive modification of the JSON file?